Tests/MSFT_xCertificateImport.tests.ps1

```diff
@@ -0,0 +1,152 @@
+            It 'should return true' {
+                ($result -is [bool]) | Should Be $true
+                $result | Should Be $true
+            }
+        }
+        Context 'a single missing file by pipeline' {
+            It 'should throw an exception' {
+                # directories are not valid
+                { $invalidPath | Validate-CertificatePath } | Should Throw
+            }
+        }
+        Context 'a single missing file by pipeline with -Quiet' {
+            $result =  $invalidPath | Validate-CertificatePath -Quiet
+            It 'should return false' {
+                ($result -is [bool]) | Should Be $true
+                $result | Should Be $false
+            }
+        }
+    }
+    Describe 'Validate-Thumbprint' {
+
+        Context 'a single valid thumbrpint by parameter' {
+            $result = Validate-Thumbprint -Thumbprint $validThumbprint
+            It 'should return true' {
+                ($result -is [bool]) | Should Be $true
+                $result | Should Be $true
+            }
+        }
+        Context 'a single invalid thumbprint by parameter' {
+            It 'should throw an exception' {
+                # directories are not valid
+                { Validate-Thumbprint -Thumbprint $invalidThumbprint } | Should Throw
+            }
+        }
+        Context 'a single invalid thumbprint by parameter with -Quiet' {
+            $result = Validate-Thumbprint $invalidThumbprint -Quiet
+            It 'should return false' {
+                ($result -is [bool]) | Should Be $true
+                $result | Should Be $false
+            }
+        }
+        Context 'a single valid thumbprint by pipeline' {
+            $result = $validThumbprint | Validate-Thumbprint
+            It 'should return true' {
+                ($result -is [bool]) | Should Be $true
+                $result | Should Be $true
+            }
+        }
+        Context 'a single invalid thumborint by pipeline' {
+            It 'should throw an exception' {
+                # directories are not valid
+                { $invalidThumbprint | Validate-Thumbprint } | Should Throw
+            }
+        }
+        Context 'a single invalid thumbprint by pipeline with -Quiet' {
+            $result =  $invalidThumbprint | Validate-Thumbprint -Quiet
+            It 'should return false' {
+                ($result -is [bool]) | Should Be $true
+                $result | Should Be $false
+            }
+        }
+    }
+    Describe 'Get-TargetResource' {
+        $null | Set-Content -Path $validPath
+
+        $result = Get-TargetResource -Thumbprint $validThumbprint -Path $validPath
+        It 'should return a hashtable' {
+            ($result -is [hashtable]) | Should Be $true
+        }
+        It 'should contain the input values' {
+            $result.Thumbprint | Should BeExactly $validThumbprint
+            $result.Path | Should BeExactly $validPath
```

Comment:
There should be a test for if get correctly returns if the certificate is already imported.